src/firmware/app/control/wheel_controller.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "firmware/shared/circular_buffer.h"
+
+// wheel_controller_difference_equation.h
+typedef struct WheelController
+{
+    CircularBuffer_t* previous_command_buffer;
+    CircularBuffer_t* previous_output_sample_buffer;
+    float* command_coefficients;
+    float* output_sample_coefficients;
+    unsigned int num_command_coefficients;
+    unsigned int num_output_sample_coefficients;
+} WheelController_t;
+
+/**
+ * Function creates a WheelController_t with the specified parameters and returns a
+ pointer to it. The wheel_controller_destroy function MUST be called before the
+ WheelController_t* goes out of scope.
+ * The WheelController_t allows previously sampled wheel states to be added to it's state,
+ and allows for new wheel commands to be requested.
+ *
+ * The coefficients parameters are in order of decreasing recentness - where
+ coefficient_array[0] is the most recent value.
+ *
+ * The convention is as follows:
+ * coeff[0] -> z^0
+ * coeff[1] -> z^-1
+ * coeff[2] -> z^-2
+ * ...
+ * coeff[N] -> z^-N
```

Comment:
Can we add a comment with a difference equation to explain how the coefficients get put into the difference equation?

Reply:
Added a comment showing how the difference equation is constructed